sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -895,6 +898,19 @@ private[hive] object HiveClientImpl {
     Option(hc.getComment).map(field.withComment).getOrElse(field)
   }
 
+  private def verifyColumnDataType(schema: StructType): Unit = {
+    schema.foreach(field => {
+      val typeString = field.dataType.catalogString
```

Comment:
`catalogString` is generated by Spark. It is not related to the restriction of Hive. 

See my fix: https://github.com/gatorsmile/spark/commit/bdcb9c8d29db022d9703eb91ef3f74c35bc24ec1

After you applying my fix, you also need to update the test cases to make the exception types consistent.

Reply:
I have taken your change and incorporated it in the latest commit.  Thanks.